Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/presencetracing/ui/poster/QrCodePosterTestViewModel.kt

```diff
@@ -84,13 +89,21 @@ class QrCodePosterTestViewModel @AssistedInject constructor(
             isRunning = true
             val traceLocation = traceLocation()
             val correctionLevel = appConfigProvider.getAppConfig().presenceTracing.qrCodeErrorCorrectionLevel
-            val qrCode = qrCodeGenerator.createQrCode(
-                input = traceLocation.locationUrl,
-                length = length,
-                margin = 0,
-                correctionLevel = correctionLevel
-            )
-            qrCodeBitmap.postValue(qrCode)
+
+            val drawable = run {
+                val req = ImageRequest.Builder(context).apply {
+                    data(
+                        CoilQrCode(
+                            content = traceLocation.locationUrl,
+                            options = QrCodeOptions(correctionLevel = correctionLevel)
+                        )
+                    )
+                    size(length)
+                }.build()
+                context.imageLoader.execute(req).drawable
+            }
+
+            qrCodeImage.postValue(drawable)
```

Comment:
This block seems to be repeated , maybe we cloud extract into a reusable function such as:
```kotlin
fun Context.loadQrcode(coilQrCode: CoilQrCode, size:Int) = run {
                 val req = ImageRequest.Builder(applicationContext).apply {
                     data(coilQrCode) 
                     size(size)
                 }.build()
                 applicationContext.imageLoader.execute(req).drawable
             }

   qrCodeImage.postValue(drawable)
}
```